precompiles/authorization/types.go

```diff
@@ -238,3 +240,48 @@ func validateMsgTypes(arg interface{}) ([]string, error) {
 
 	return typeURLs, nil
 }
+
+// ConvertToAllocation converts the transfer types Allocation to the ICS20 Allocation.
+func convertToAllocation(allocs []transfertypes.Allocation) []cmn.Allocation {
+	// Convert to Allocations to emit the IBC transfer authorization event
+	allocations := make([]cmn.Allocation, len(allocs))
+	for i, a := range allocs {
+		spendLimit := make([]cmn.Coin, len(a.SpendLimit))
+		for j, c := range a.SpendLimit {
+			spendLimit[j] = cmn.Coin{
+				Denom:  c.Denom,
+				Amount: c.Amount.BigInt(),
+			}
+		}
+
+		allocations[i] = cmn.Allocation{
+			SourcePort:    a.SourcePort,
+			SourceChannel: a.SourceChannel,
+			SpendLimit:    spendLimit,
+			AllowList:     a.AllowList,
+		}
```

Comment:
This is just a suggestion to make more explicit what is an index and what is the type

```suggestion
	for i, allocation := range allocs {
		spendLimit := make([]cmn.Coin, len(a.SpendLimit))
		for j, coin := range a.SpendLimit {
			spendLimit[j] = cmn.Coin{
				Denom:  coin.Denom,
				Amount: coin.Amount.BigInt(),
			}
		}

		allocations[i] = cmn.Allocation{
			SourcePort:    allocation.SourcePort,
			SourceChannel: allocation.SourceChannel,
			SpendLimit:    spendLimit,
			AllowList:     allocation.AllowList,
		}
```